File: java/json-transform/src/main/java/co/nlighten/jsontransform/functions/TransformerFunctionSplit.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.functions.common.*;

import java.util.Map;

public class TransformerFunctionSplit extends TransformerFunction {
    public TransformerFunctionSplit() {
        super(FunctionDescription.of(
                Map.of(
                        "delimiter", ArgumentType.of(ArgType.String).position(0).defaultString(""),
                        "limit", ArgumentType.of(ArgType.Integer).position(1).defaultInteger(0)
                )
        ));
    }
    @Override
    public Object apply(FunctionContext context) {
        var str = context.getString(null);
        if (str == null) {
            return null;
        }
        return str.split(context.getString("delimiter"), context.getInteger("limit"));
    }
}

File: javascript/json-transform/src/functions/TransformerFunctionSplit.ts
import TransformerFunction from "./common/TransformerFunction";
import { ArgType } from "./common/ArgType";
import FunctionContext from "./common/FunctionContext";
import { createComparator, getAsString, isNullOrUndefined } from "../JsonHelpers";
import CompareBy from "./common/CompareBy";
import JsonElementStreamer from "../JsonElementStreamer";
import javaSplit from "./utils/javaSplit";

class TransformerFunctionSplit extends TransformerFunction {
  constructor() {
    super({
      arguments: {
        delimiter: { type: ArgType.String, position: 0, defaultString: "" },
        limit: { type: ArgType.Integer, position: 1, defaultInteger: 0 },
      },
    });
  }

  override async apply(context: FunctionContext): Promise<any> {
    const str = await context.getString(null);
    if (str == null) {
      return null;
    }
    const delimiter = (await context.getString("delimiter")) ?? "";
    const limit = await context.getInteger("limit");
    return javaSplit(str, delimiter, limit);
  }
}

export default TransformerFunctionSplit;
